src/main/kotlin/org/b12x/gfe/core/controller/version/LocalVersions.kt

```diff
@@ -0,0 +1,95 @@
+package org.b12x.gfe.core.controller.version
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.b12x.gfe.core.model.DataDownload
+import org.b12x.gfe.core.model.parsers.ParserVersionData
+import org.b12x.gfe.utilities.DirectoryManagement
+import org.b12x.gfe.utilities.FileManagement
+import java.io.File
+import java.nio.file.Paths
+import kotlin.io.path.exists
+
+class LocalVersions(loci: String) {
+
+    private val directoryManagement = DirectoryManagement()
+    private val gsgDataLocation = directoryManagement.setLociLocation(loci)
+    private val headerLength = 50  // bytes
+
+    val versionsList = createVersions()
+
+    /**
+     * Gets all versions for a specific set of genes in the user's GSG data directory.
```

Comment:
I don't understand what versions are, but maybe it would be obvious to someone familiar with your domain.

Reply:
Okay, the core version files are the disaster I'm currently working with. The basic concept is every 3 months a new version (list of confirmed variants of the genes) is released. So not only do you need to select which gene you're looking at, but which version of it you're looking at. As sequencing technology gets better and cheaper they're better able to confirm DNA sequences, but that means sometimes a gene variant they thought they knew might change. So knowing which version you're looking at matters for research.

This, combined with my boss' request that the program be useable without internet, has made things extremely complex. To do this I have a local file structure that includes version folders, with data files inside corresponding to each version. To be useable without internet you have to download the data files before hand. (They're CSV files under 1 MB each, so it's not a big deal to download.)

Anyway, KIR and HLA each use a different set of version numbers. And as they update quarterly, and the database holding them only keeps the three most recent versions, you ended up with downloaded legacy data. To use that legacy data you have to look and see what downloaded versions you have, as well as which specific genes you're looking at. So versions become important again as you might only have half the available gene data, so you have to go through each version and check what's actually there. 

This has been a huge process to work out, changing from KIR to HLA or vice versa (the loci choicebox) means you have to update the version choicebox accordingly, and depending on if the version is legacy or not, (Or you don't have internet access,) you have to check to see what individual genes you actually have data for, and only display those (that's not working all the way yet.)